Update airflow-datasets.md with note about new listeners (#3188)
astronomer/docs: learn/airflow-datasets.md

- Get better visibility into how your DAGs are connected and how they depend on data. The **Datasets** tab in the Airflow UI shows a graph of all dependencies between DAGs and datasets in your Airflow environment.
- Reduce costs, because datasets do not use a worker slot in contrast to sensors or other implementations of cross-DAG dependencies.

> :::note Listener Hooks

Reviewer: ```suggestion
:::note Listening for dataset changed
```

> As of Airflow 2.8, you can use [Listeners](http://apache-airflow-docs.s3-website.eu-central-1.amazonaws.com/docs/apache-airflow/stable/administration-and-deployment/listeners.html#listeners) to enable Airflow to notify you when certain dataset events occur. There are two Listener hooks for the following events: 

Reviewer: ```suggestion
As of Airflow 2.8, you can use [listeners](https://airflow.apache.org/docs/apache-airflow/stable/administration-and-deployment/listeners.html#listeners) to enable Airflow to notify you when certain dataset events occur. There are two listener hooks for the following events: 
```

- on_dataset_created
- on_dataset_changed

> For examples, please refer to the following document: [Listener Plugin of Airflow](http://apache-airflow-docs.s3-website.eu-central-1.amazonaws.com/docs/apache-airflow/stable/howto/listener-plugin.html)

Reviewer: ```suggestion
For examples, refer to our [Create Airflow listeners tutorial](https://docs.astronomer.io/learn/airflow-listeners).
```
